public/messages.proto

```diff
@@ -1,700 +1,700 @@
-    }
-}
-
-message ServerRoomFlagsUpdated {
-    repeated GameFlag flags = 0;
-    message GameFlag {
-        required string flag = 0;
-        optional int32 int_val = 1;
-        optional string str_val = 2;
-        optional float float_val = 3;
-        optional bool bool_val = 4;
-        optional uint32 u_int_val = 5;
-    }
-}
-
-message ServerRoomFlagsUpdateFailed {
-    string reason = 0;
-}
-
-message ClientJoinRoom {
-    required string id = 0;
-    optional string password = 1;
-}
-
-message ServerJoinRoomSuccess {
-    required string id = 0;
-    required string name = 1;
-    required uint32 gamemode = 2;
-    required uint32 max_users = 3;
-    optional string password = 4;
-    required bool locked = 5;
-    repeated User users = 6;
-
-    message User {
-        required string user_id = 0;
-        required string name = 1;
-        required bool ready = 2;
-        required bool owner = 3;
-    }
-}
-
-message ServerJoinRoomFailed {
-    string reason = 0;
-}
-
-message ServerUserJoinedRoom {
-    required string user_id = 1;
-    required string name = 2;
-}
-
-message ClientLeaveRoom {
-    required string user_id = 1; //should be empty msg
-}
-
-message ServerUserLeftRoom {
-    required string user_id = 1;
-}
-
-message ClientKickUser {
-    required string user_id = 1;
-}
-
-message ServerUserKicked {
-    required string user_id = 1;
-}
-
-message ClientBanUser {
-    required string user_id = 1;
-}
-
-message ServerUserBanned {
-    required string user_id = 1;
-}
-
-message ClientReadyState {
-    required bool ready = 0;
-    optional string seed = 1;
-    repeated string mods = 2;
-    optional string version = 3;
-    optional bool beta = 4;
-}
-
-message ServerUserReadyState {
-    required string user_id = 0;
-    required bool ready = 1;
-    optional string seed = 2;
-    repeated string mods = 3;
-    optional string version = 4;
-    optional bool beta = 5;
-}
-
-message ClientStartRun {
-    required bool forced = 0;
-}
-
-message ServerHostStart {
-    required bool forced = 0;
-}
-
-message ClientRequestRoomList {
-    required uint32 page = 0;
-}
-
-message ServerRoomList {
-    repeated Room rooms = 0;
-    optional uint32 pages = 1;
-    message Room {
-        required string id = 0;
-        required string name = 1;
-        required uint32 gamemode = 2;
-        required uint32 cur_users = 3;
-        required uint32 max_users = 4;
-        required bool protected = 5;
-        required string owner = 6;
-        required bool locked = 7;
-    }
-}
-
-message ServerRoomAddToList {
-    required Room room = 0;
-    message Room {
-        required string id = 0;
-        required string name = 1;
-        required uint32 gamemode = 2;
-        required uint32 cur_users = 3;
-        required uint32 max_users = 4;
-        required bool protected = 5;
-        required string owner = 6;
-        required bool locked = 7;
-    }
+syntax = "proto3";
+package NT;
```

Comment:
What has changed in the protobuf file?
Well, first I had to move the syntax statement to the first line, so that it would get parsed at all.
Then, I had to shift nearly every single identifier by 1, so that nothing starts with zero.
And then I removed every single `required`, since proto3 apparently doesn't support those anymore.

Reply:
The changes have been moved to a `messages-updated.proto` file. ^